### ISSUE TITLE
Use Dockerfile file paths when using Sync

### ISSUE THREAD

**Issue by alewis001** [GoogleContainerTools/skaffold#1581]
### Desired Behavior
Files matching the sync pattern are copied to their locations as defined in the Dockerfile.

E.g.

`Dockerfile`:
```
# Start with OL runtime.
FROM open-liberty:kernel

# Configure the features and JVM settings for the Server
COPY server.xml /config/
COPY jvm.options /config/

# Copy shared/sttic libraries into a global location.
COPY lib /config/lib/global

# Add the Azure Certificate for Redis to enable HTTPS connection. 
COPY azure-redis.cer /opt/ol/wlp/output/defaultServer/resources/security/
WORKDIR /opt/ol/wlp/output/defaultServer/resources/security/
RUN keytool -noprompt -importcert -file azure-redis.cer -alias azure-redis -keystore key.jks -storepass Liberty -storetype jks

# Copy in the application
COPY myapp.war /config/dropins
```

In my particular case, as OpenLiberty supports dynamic updates of the `server.xml` and any wars deployed to it, I believe it would be useful to have the sync feature be able to copy those files to the active container based on their location as specified in the Dockerfile rather than having to duplicate the "config" in two places.

I'm able to get the functionality I'm looking for with Skaffold 0.22.0 but I currently need to:
* Update the Dockerfile to include `WORKDIR /` at the end of the file.
* Set the sync targets in `skaffold.yaml` to be the same paths as defined in the Dockerfile.

**Comment by r2d4** [comment]
+1 ref https://github.com/GoogleContainerTools/skaffold/issues/1166

**Comment by ketaanhshah** [comment]
Hey guys, I am building a Docker container with Open Liberty and I used the following in my DockerFile to copy the Azure Cert for HTTPS.

# Add the Azure Certificate for Redis to enable HTTPS connection. 
COPY azurewebsites.cer /opt/ol/wlp/output/defaultServer/resources/security/
WORKDIR /opt/ol/wlp/output/defaultServer/resources/security/
RUN keytool -noprompt -importcert -file azurewebsites.cer -alias azureazurewebsites -keystore azurewebsites.jks -storepass Liberty -storetype jks

but I am getting this error:- 
Step 6/6 : RUN keytool -noprompt -importcert -file azurewebsites.cer -alias azurewebsites -keystore azurewebsites.jks -storepass Liberty -storetype jks
 ---> Running in 8acb3a438510
Certificate was added to keystore
keytool error (likely untranslated): java.io.FileNotFoundException: azurewebsites.jks (Permission denied)
The command '/bin/sh -c keytool -noprompt -importc

Anyone know how I can fix this error so I can generate my docker.

**Comment by alewis001** [comment]
Hi,

Have you got a file called azurewebsites.jks? I think the default in OL the
file is called key.jks. does it work if you swap azurewebsites.jks for
key.jks in the keytool command?

Cheers

**Comment by ketaanhshah** [comment]
Thank you Alex, No I only downloaded the Azurewebsites.cer (as Base64) from browser and then thought running the following should help me generate the .jks file and I thought I can give it a name Azurewebsites.jsk. Am I doing something completely wrong here?

I am trying to generate an Open Liberty Docker container and using it in Azure web-app and whcih has to talk to another Azure web api (https) and it is failing currently with error of (ROR   ] CWPKI0022E: SSL HANDSHAKE FAILURE:  A signer with SubjectDN CN=*.azurewebsites.net was sent from the target host.  The signer might need to be added to local trust store /opt/ol/wlp/output/defaultServer/resources/security/key.p12).

While searching all places I came across your post and I thought this was it how I can import the Azurewebsites cer to my docker container I am building for OL.

**Comment by ketaanhshah** [comment]
hey Alex,
I think that was it I had to use key.jks and it resolved my issue, Thank you !

**Comment by alewis001** [comment]
Great, glad it worked!

Just in case it helps and as this doesn't really have anything to do with
Skaffold, you'd probably be best asking questions on either the Openliberty
Google Group or their Stackoverflow: https://openliberty.io/support/

Cheers

**Comment by ketaanhshah** [comment]
Thank you Alex, your POST above was GREAT HELP to me! I will make sure to post questions on appropriate forums but your DOCKER FILE post above helped me tons! Have a great weekend!

**Comment by tejal29** [comment]
@alewis001 this is be fixed now with @2088 and #1812

**Comment by alewis001** [comment]
Sure, I will attempt to verify this as soon as I can.

**Comment by tejal29** [comment]
This is fixed in #2008 and #1812. Please re-open if you still see this.
